relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,78 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        // TODO: make keys lazy init
+        let mut keys = Vec::new();
+
+        // collect the variables keys and scrub them out
+        if let Some(request) = event.request.value_mut() {
+            if let Some(Value::Object(data)) = request.data.value_mut() {
+                if let Some(Annotated(Some(Value::Bool(graph_ql_error)), _)) =
+                    request.other.get("graphql_error")
+                {
+                    dbg!("graphql_error request {graphql_error}");
+                    if *graph_ql_error {
+                        if let Some(Annotated(Some(Value::Object(variables)), _)) =
+                            data.get_mut("variables")
+                        {
+                            for item in variables.iter_mut() {
+                                keys.push(item.0.to_string());
+                                item.1
+                                    .set_value(Some(Value::String("[Filtered]".to_string())));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // scrub PII from the data object if they match the variables keys
+        if let Some(contexts) = event.contexts.value_mut() {
+            if let Some(Context::Response(response)) = contexts.get_context_mut("response") {
+                if let Some(Value::Object(data)) = response.data.value_mut() {
+                    if let Some(Annotated(Some(Value::Bool(graph_ql_error)), _)) =
+                        response.other.get("graphql_error")
+                    {
+                        if *graph_ql_error {
+                            if let Some(Annotated(Some(Value::Object(response_data)), _)) =
+                                data.get_mut("data")
+                            {
+                                scrub_data_from_response(keys.clone(), response_data);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+}
+
+fn scrub_data_from_response(keys: Vec<String>, data: &mut BTreeMap<String, Annotated<Value>>) {
```

Comment:
It should be possible to pass a `BTreeSet<&str>` instead of a `Vec<String>` here to avoid cloning.
```suggestion
fn scrub_data_from_response<'a>(keys: BTreeSet<&'a str>, data: &mut BTreeMap<String, Annotated<Value>>) {
```